R/database.R

```diff
@@ -2628,3 +2636,258 @@ get_gene_expression = function(metadata,
+#'
+get_manta_sv_by_sample = function(this_sample_id,
+                                  these_samples_metadata,
+                                  force_lift = TRUE,
+                                  min_vaf = 0.1,
+                                  min_score = 40,
+                                  pass = TRUE,
+                                  with_chr_prefix = FALSE,
+                                  projection = "grch37"){
+  
+  #check remote configuration
+  remote_session = check_remote_configuration(auto_connect = TRUE)
+  
+  these_samples_metadata = dplyr::filter(these_samples_metadata, sample_id == this_sample_id)
+  if(!nrow(these_samples_metadata==1)){
+    stop("metadata does not seem to contain your this_sample_id or you didn't provide one")
+  }
+  
+  #get wildcards
+  tumour_sample_id = this_sample_id
+  unix_group = pull(these_samples_metadata, unix_group)
+  seq_type = pull(these_samples_metadata, seq_type)
+  genome_build = pull(these_samples_metadata, genome_build)
+  pairing_status = pull(these_samples_metadata, pairing_status)
+  
+  if(pairing_status == "matched"){
+    normal_sample_id = pull(these_samples_metadata, normal_sample_id)
+  }else{
+    normal_sample_id = config::get("unmatched_normal_ids")[[unix_group]][[seq_type]][[genome_build]]
+  }
+
+  #get samples from individual flat files
+  path_template = config::get("results_flatfiles")$sv_manta$template
+  
+  if(!remote_session){
+    path_template_full = paste0(config::get("project_base"), path_template)
+    bedpe_path = glue::glue(path_template_full)
+    if(!file.exists(bedpe_path)){
+      print(paste("missing: ", bedpe_path))
+      message("Cannot find file locally. If working remotely, perhaps you forgot to load your config (see below) or sync your files?")
+      message('Sys.setenv(R_CONFIG_ACTIVE = "remote")')
+    }
+  }else{
+    local_path_template = paste0(config::get("project_base", config = "remote"), path_template)
+    bedpe_path = glue::glue(local_path_template)
+      
+    #check if the requested file is on local machine, if not, get it!
+    if(!file.exists(bedpe_path)){
+      remote_path_template = paste0(config::get("project_base", config = "default"), path_template)
+      remote_bedpe_path = glue::glue(remote_path_template)
+      cat(paste0("Local file not found.\ntrying to copy requested file: ", remote_bedpe_path, "\n", "To: ", bedpe_path))
+      dirN = dirname(bedpe_path)
+      suppressMessages(suppressWarnings(dir.create(dirN, recursive = T)))
+      ssh::scp_download(ssh_session, remote_bedpe_path, dirN)
+    }
+  }
+  
+  #read sample flat-file
+  message(paste0("Reading ", this_sample_id, " from: ", bedpe_path))
+  bedpe_dat_raw = suppressMessages(read_tsv(bedpe_path, comment = "##"))
+    
+  #if returned manta calls data frame is empty, return and potentially move over to the next sample (if get_manta_sv_by_samples is called).
```

Comment:
This is a bit of a confusing comment at the end. The behaviour of what happens next is not dependent on this function so I'd just leave it as "return empty data frame" or something like that

Reply:
Sounds good to me!

Reply:
is this resolved?